pom.xml

```diff
@@ -494,62 +486,6 @@
         </plugins>
       </build>
     </profile>
-
```

Comment:
We do not need to remove this profile. It is still needed and works normally 

Changes you made at StepDefinisions.java [1] is sufficient to fulfill the comment 

[1] - https://github.com/open-feature/java-sdk/pull/546/files#diff-ca6f171ac07058b6e01935480aefe8e8a07379832fc06c334a9ad8887ae3881f

Reply:
Without this section, we do not initialize the test-harness submodule. See the build result  - https://github.com/open-feature/java-sdk/actions/runs/5824793560/job/15795407914?pr=546

Reply:
By this comment it can be removed, if you still think it should remain update here and I can return it.
```
<!-- this profile handles running the flagd e2e tests -->
      <!-- TODO: this profile can likely be removed with TODO: this section should be updated with https://github.com/open-feature/java-sdk/issues/523 -->
      <!-- TODO: we should pull the submodule and run these tests unconditionall once flagd isn't required -->
```

Reply:
yes, we still need them. Tests are now independent from flagd but we need this profile to intialize git submodule and run gherkin tetsts

See the GitHub action - https://github.com/open-feature/java-sdk/blob/main/.github/workflows/pullrequest.yml#L42-L43

Reply:
I was thinking we could just always pull the submodule and copy the gherkin tests, since now they don't require the external flagd process. I dont really mind if we keep these integration tests on a separate profile or not.

I think if we keep the profile though, we should add back the little bit of documentation in the contributing guide about it.